Ominous linker messages when using platform blackpillv2

What I did:
```
git clone git@github.com:blackmagic-debug/blackmagic.git
cd blackmagic
git submodule update --init --recursive
make PROBE_HOST=blackpillv2
```
HEAD was at 19d9111a834e53b0be6bda1b1294c03a7f31431c and the actual gcc-arm-none-eabi-12.2.mpacbti-bet1 compiler from ARM was used on Linux.

These commands resulted in the following linker warnings:
```
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-closer.o): in function `_close_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/closer.c:47: warning: _close is not implemented and will always fail
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-signalr.o): in function `_getpid_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/signalr.c:83: warning: _getpid is not implemented and will always fail
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-signalr.o): in function `_kill_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/signalr.c:53: warning: _kill is not implemented and will always fail
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-lseekr.o): in function `_lseek_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/lseekr.c:49: warning: _lseek is not implemented and will always fail
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-readr.o): in function `_read_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/readr.c:49: warning: _read is not implemented and will always fail
/home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /home/stefan/opt/gcc-arm-none-eabi-12.2.mpacbti-bet1/bin/../lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libc.a(libc_a-writer.o): in function `_write_r':
/data/jenkins/workspace/GNU-toolchain/arm-12-mpacbti/src/newlib-cygwin/newlib/libc/reent/writer.c:49: warning: _write is not implemented and will always fail
```
I don't have my ordered BlackPill yet so I can't test but linker warnings regarding newlib-cygwin make me suspicious. Is this supposedly so?

--- a/src/platforms/common/usb_serial.c
+++ b/src/platforms/common/usb_serial.c
@@ -40,6 +40,12 @@
  *
  */
 
+#ifndef ENABLE_DEBUG
+#include <sys/stat.h>
+#include <string.h>
+
+typedef struct stat stat_s;
+#endif
 #include "general.h"
 #include "gdb_if.h"
 #include "usb_serial.h"
@@ -422,5 +428,68 @@ void debug_monitor_handler(void)
 		frame->r0 = UINT32_MAX;
 	}
 	__asm__("bx lr");
+}
+#else
+/* This defines stubs for the newlib fake file IO layer for compatability with GCC 12 `-spec=nosys.spec` */
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _write(const int file, const void *const buffer, const size_t length)
+{
+	(void)file;
+	(void)buffer;
+	return length;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _read(const int file, void *const buffer, const size_t length)
+{
+	(void)file;
+	(void)buffer;
+	return length;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+off_t _lseek(const int file, const off_t offset, const int direction)
+{
+	(void)file;
+	(void)offset;
+	(void)direction;
+	return 0;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _fstat(const int file, stat_s *stats)
+{
+	(void)file;
+	memset(stats, 0, sizeof(*stats));
+	return 0;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _isatty(const int file)
+{
+	(void)file;
+	return true;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _close(const int file)
+{
+	(void)file;
+	return 0;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+pid_t _getpid(void)
+{
+	return 1;
+}
+
+/* NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp) */
+int _kill(const int pid, const int signal)
+{
+	(void)pid;
+	(void)signal;
+	return 0;
 }
 #endif

--- a/src/target/nxpke04.c
+++ b/src/target/nxpke04.c
@@ -114,7 +114,7 @@
 static const uint8_t cmd_lens[] = {4, 1, 2, 3, 6, 0, 6, 6, 1, 2, 2, 1, 5, 3, 3};
 
 /* Flash routines */
-static bool ke04_command(target_s *t, uint8_t cmd, uint32_t addr, const uint8_t *data);
+static bool ke04_command(target_s *t, uint8_t cmd, uint32_t addr, const void *data);
 static bool ke04_flash_erase(target_flash_s *f, target_addr_t addr, size_t len);
 static bool ke04_flash_write(target_flash_s *f, target_addr_t dest, const void *src, size_t len);
 static bool ke04_flash_done(target_flash_s *f);
@@ -260,7 +260,7 @@ static bool ke04_wait_complete(target_s *t)
 	return true;
 }
 
-static bool ke04_command(target_s *t, uint8_t cmd, uint32_t addr, const uint8_t *data)
+static bool ke04_command(target_s *t, uint8_t cmd, uint32_t addr, const void *const data)
 {
 	/* Set FCLKDIV to 0x17 for 24MHz (default at reset) */
 	uint8_t fclkdiv = target_mem_read8(t, FTMRE_FCLKDIV);
@@ -296,9 +296,10 @@ static bool ke04_command(target_s *t, uint8_t cmd, uint32_t addr, const uint8_t 
 	}
 
 	/* Write one or two 32 bit words of data */
-	for (uint8_t offset = 0; fccob_idx < cmd_len; ++fccob_idx, offset += 2U) {
+	const uint16_t *const cmd_data = (const uint16_t *)data;
+	for (uint8_t offset = 0; fccob_idx < cmd_len; ++fccob_idx, ++offset) {
 		target_mem_write8(t, FTMRE_FCCOBIX, fccob_idx);
-		target_mem_write16(t, FTMRE_FCCOB, *(uint16_t *)(data + offset));
+		target_mem_write16(t, FTMRE_FCCOB, cmd_data[offset]);
 	}
 
 	/* Enable execution by clearing CCIF */
@@ -354,5 +355,5 @@ static bool ke04_flash_done(target_flash_s *f)
 	/* Note: Cumulative programming is not allowed according to the RM */
 	uint32_t vals[2] = {target_mem_read32(f->t, FLASH_SECURITY_WORD_ADDRESS), 0};
 	vals[0] = (vals[0] & 0xff00ffffU) | (FLASH_SECURITY_BYTE_UNSECURED << 16U);
-	return ke04_command(f->t, CMD_PROGRAM_FLASH_32, FLASH_SECURITY_WORD_ADDRESS, (uint8_t *)&vals);
+	return ke04_command(f->t, CMD_PROGRAM_FLASH_32, FLASH_SECURITY_WORD_ADDRESS, &vals);
 }

